src/mono/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeTypeBuilder.Mono.cs

```diff
@@ -430,7 +430,7 @@ protected override void AddInterfaceImplementationCore([DynamicallyAccessedMembe
                 if (types == null)
                 {
                     if (count > 1)
-                        throw new AmbiguousMatchException();
+                        ThrowHelper.ThrowAmbiguousMatchException(found!);
```

Comment:
`found` is known non null here, but the compiler doesn't grok that. It comes from the `foreach (` over `cb`  so the `!` is warranted.

Could do a tiny bit of cleanup here to make that more obvious to the compiler (and subsequent developers)